app/jobs/assignment_repo/create_github_repository_job.rb

```diff
@@ -16,22 +16,39 @@ class CreateGitHubRepositoryJob < ApplicationJob
     #
     # rubocop:disable MethodLength
     # rubocop:disable AbcSize
-    # rubocop:disable CyclomaticComplexity
-    # rubocop:disable PerceivedComplexity
     def perform(assignment, user, retries: 0)
       start = Time.zone.now
-
       invite_status = assignment.invitation.status(user)
-
-      return unless invite_status.waiting? || invite_status.errored_creating_repo?
+      return unless invite_status.waiting?
```

Comment:
We actually only want to kick off a job if the `invite_status` is `waiting`. `waiting` should be the only acceptable state for job execution. 

If the status is `errored_creating_repo` it would be changed to `waiting` in `AssignmentInvitationsController#create_repo`.